astor/code_gen.py

```diff
@@ -50,22 +50,30 @@ def to_source(node, indent_with=' ' * 4, add_line_information=False,
                                 pretty_string)
     generator.visit(node)
     generator.result.append('\n')
-    return pretty_source(str(s) for s in generator.result)
+    if set(generator.result[0]) == set('\n'):
+        generator.result[0] = ''
```

Comment:
This removes extra linefeeds at the start of the result.  This is more performant than the old method, which required a conditional check on every write to see if we were at the beginning of the result.